### PR TITLE
Add troubleshooting section to docs with an entry on Spark issues

### PR DESCRIPTION
**Pull request by jfnavarro**
# nf-core/sarek pull request

Many thanks for contributing to nf-core/sarek!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

## PR checklist

- [x] Documentation in `docs` is updated

**Learn more about contributing:** [CONTRIBUTING.md](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -3,8 +3,8 @@
 The nf-core/sarek documentation is split into the following pages:
 
 - [Usage](usage.md)
-  - An overview of how the pipeline works, how to run it and a description of all of the different command-line flags.
+  - An overview of how the pipeline works, how to run it, a description of all of the different command-line flags and solutions to common issues.
 - [Output](output.md)
   - An overview of the different results produced by the pipeline and how to interpret them.
-
+
```
Comment (MaxUlysse) [inline docs/README.md:9]:
```suggestion
```

File: docs/usage.md [hunk 2]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
 Default: `/home/ec2-user/miniconda/bin/aws`.
 
 Please make sure to also set the `-w/--work-dir` and `--outdir` parameters to a S3 storage bucket of your choice - you'll get an error message notifying you if you didn't.
+
+## Troubleshooting
+
```
Comment (MaxUlysse) [inline docs/usage.md:1744]:
```suggestion
```

File: docs/usage.md [hunk 3]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
...
+Edit the file ```/etc/sysconfig/docker```and add the new limits to OPTIONS like this:
+
+```bash
+OPTIONS=”—default-ulimit nofile=65535:65535"
+```
+
+Re-start your session. 
+
+Note that the way to increase the open file limit in your system may be slightly different
+or require additional steps.
```
Comment (MaxUlysse) [inline docs/usage.md:1780]:
```suggestion
or require additional steps.

```

**Review by jfnavarro — COMMENTED** [review]
Looks good to me.

File: docs/usage.md [hunk 4]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
 Default: `/home/ec2-user/miniconda/bin/aws`.
 
 Please make sure to also set the `-w/--work-dir` and `--outdir` parameters to a S3 storage bucket of your choice - you'll get an error message notifying you if you didn't.
+
+## Troubleshooting
+
+### Spark related issues
+
+If you have problems running processes that make use of Spark such as ```MarkDuplicates```.
```
Comment (MaxUlysse) [inline docs/usage.md:1747]:
```suggestion
If you have problems running processes that make use of Spark such as ```MarkDuplicates```.
```

File: docs/usage.md [hunk 5]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
...
 
 Please make sure to also set the `-w/--work-dir` and `--outdir` parameters to a S3 storage bucket of your choice - you'll get an error message notifying you if you didn't.
+
+## Troubleshooting
+
+### Spark related issues
+
+If you have problems running processes that make use of Spark such as ```MarkDuplicates```. 
+You are probably experiencing issues with the limit of open files in your system. You can
+check your current limit by typing the following:
```
Comment (MaxUlysse) [inline docs/usage.md:1749]:
```suggestion
You are probably experiencing issues with the limit of open files in your system.
You can check your current limit by typing the following:
```

File: docs/usage.md [hunk 6]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
...
+
+```bash
+ulimit -n
+```
+
+The default limit size is usually 1024 which is quite low to run Spark jobs. In order to increase
+the size limit permantly you can:
```
Comment (MaxUlysse) [inline docs/usage.md:1756]:
```suggestion
The default limit size is usually 1024 which is quite low to run Spark jobs.
In order to increase the size limit permanently you can:
```

File: docs/usage.md [hunk 7]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
...
+Edit the file ```/etc/sysconfig/docker``` and add the new limits to OPTIONS like this:
+
+```bash
+OPTIONS=”—default-ulimit nofile=65535:65535"
+```
+
+Re-start your session. 
+
+Note that the way to increase the open file limit in your system may be slightly different
+or require additional steps.
```
Comment (MaxUlysse) [inline docs/usage.md:1780]:
```suggestion
Note that the way to increase the open file limit in your system may be slightly different or require additional steps.
```

File: docs/usage.md [hunk 8]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
...
+```
+
+The default limit size is usually 1024 which is quite low to run Spark jobs. In order to increase
+the size limit permantly you can:
+
+Edit the file ```/etc/security/limits.conf``` and add the lines:
+
+```bash
+*     soft   nofile  65535
+*     hard   nofile  65535
```
Comment (MaxUlysse) [inline docs/usage.md]:
```suggestion
*     hard   nofile  65535
```

File: docs/usage.md [hunk 9]
```diff
@@ -1737,3 +1739,42 @@ The [AWS CLI](https://www.nextflow.io/docs/latest/awscloud.html#aws-cli-installa
...
+fs.file-max = 65535
+```
+
+Edit the file ```/etc/sysconfig/docker``` and add the new limits to OPTIONS like this:
+
+```bash
+OPTIONS=”—default-ulimit nofile=65535:65535"
+```
+
+Re-start your session.
```
Comment (MaxUlysse) [inline docs/usage.md]:
```suggestion
Re-start your session.
```

**Review by MaxUlysse — APPROVED** [review]
I think we're good with markdownlint now.

Thanks @jfnavarro for the PR.

That's exactly what I had in mind.

**Pull request merged**